src/envoy/http/service_control/client_cache.cc

```diff
@@ -292,8 +302,10 @@ CancelFunc ClientCache::callCheck(
                                   const std::string& body) {
           Status final_status = processScCallTransportStatus<CheckResponse>(
               status, response, body);
-          ServiceControlFilterStats::collectCallStatus(filter_stats_.check_,
-                                                       final_status.code());
+          if (!destruct_mode_) {
+            ServiceControlFilterStats::collectCallStatus(filter_stats_.check_,
```

Comment:
How about moving collectCallStatus as ClientCache class member,  and do the destruct_mode checking in the function?